imsms_analysis/analysis_runner.py

```diff
@@ -73,6 +77,105 @@ def run(self, factory):
+# https://waterprogramming.wordpress.com/2013/01/24/pbs-job-submission-with-python/
+# ^ for submitting jobs with python
+class ParallelRunner:
+    def __init__(self):
+        self.callbacks = AnalysisCallbacks()
+
+    def run(self, factory):
+        factory.validate()
+        _check_unique_names(factory)
+
+        configs = []
+        for config in factory.gen_configurations():
+            configs.append(config)
+
+        # Run test
+        configs = [x for x in factory.gen_configurations()]
+
+        self.callbacks.batch_info(configs)
+        for config in configs:
+            try:
+                print(config.analysis_name)
+                
+                # Run this custom preprocessing
+                (
+                final_biom, 
+                target, 
+                _, _
+                ) = run_preprocessing(config, self.callbacks)
+
+                base_dir = "dataset"
+                dataset = "imsms-mlab"
+                preparation = config.analysis_name
+                target_name = "disease_binary"
+                algorithm = config.mlab_algorithm
+                if algorithm is None:
+                    algorithm = "RandomForestClassifier"
+
+                # Create the  expected file structure
+                results_dir = path.join(
+                    base_dir, dataset, preparation, target_name
+                )
+                if not path.exists(results_dir):
+                    makedirs(results_dir)
+                
+                # Save table and metadata
+                table_fp = path.join(
+                    base_dir, dataset, preparation, target_name,
+                    "filtered_table.qza"
+                )
+                table_artifact = Artifact.import_data(
+                    "FeatureTable[Frequency]", 
+                    final_biom
+                )
+                table_artifact.save(table_fp)
+
+                metadata_fp = path.join(
+                    base_dir, dataset, preparation, target_name,
+                    "filtered_metadata.qza"
+                )
+                metadata_artifact = Artifact.import_data(
+                    "SampleData[Target]", target
+                )
+                metadata_artifact.save(metadata_fp)
+                
+
+                # run job indices 1...10 inclusive, letting 5 jobs run at once
+                # each job has 100 parameter sets, for a total of 1000 parameter sets
+                start = 1
+                end = 10
+                n_concurrent_jobs = 5
+                chunk_size = 100
+                (
+                    script_fp,
+                    params_fp,
+                    run_info_fp,
+                ) = orchestrate_hyperparameter_search(
+                    dataset=dataset,
+                    preparation=preparation,
+                    target=target_name,
+                    algorithm=algorithm,
+                    repeats=3,  # num CV repeats
```

Comment:
I thought we used to use 5?  Or am I thinking of the relative sizes of training and validation sets?  Is this a change from what we use in SerialRunner or is it exactly the same?

Reply:
We do a 5-fold cross validation (20-80 split), which might be what you're thinking of. Repeats just means we do this 3 times with  different partitions, for a total of 15 folds.